leader/leader_test.go

```diff
@@ -209,4 +209,137 @@ var _ = Describe("Leader election", func() {
 			Expect(pod.TypeMeta.Kind).To(Equal("Pod"))
 		})
 	})
+
```

Comment:
PS.: see that is not all changes covered by the tests. its % decreased.

Reply:
@camilamacedo86 i will enhance the coverage in the new commit. Do i need to make a new pull-request?

Reply:
@kasonglee I see you already added tests that is just fine. There is no need for a new commit.